static/js/popovers.js

```diff
@@ -383,6 +383,15 @@ exports.register_click_handlers = function () {
         e.preventDefault();
     });
 
+    $('body').on('click', '.sidebar-popover-mark-topic-read', function (e) {
+        var topic = $(e.currentTarget).attr('data-topic-name');
+        var stream = $(e.currentTarget).attr('data-stream-name');
+        popovers.hide_stream_sidebar_popover();
```

Comment:
Empirically this doesn't correctly hide the topic sidebar -- I think you want to be calling `popovers.hide_topic_sidebar_popover();`